### ISSUE TITLE
VirtualMachineScaleSet capacity is unstable when creating two of them asynchronously

### ISSUE THREAD

**Issue by tgc123** [Azure/azure-sdk-for-java#1525]
When creating two VirtualMachineScaleSets in async with a different capacity, the first one will be created with the same capacity as the latter one.

The code snippet below should create ScaleSet "test1" with capacity of 1 VM and ScaleSet "test2" with capacity of 2 VMs. However, both of them are created with 2 VMs.

Sleeping (for example 0.5s) between the requests "fixes" the issue.

```
Vector<Future<VirtualMachineScaleSet>> futures = new Vector<>();

Observable<Indexable> vms = azure.virtualMachineScaleSets()
  .define("test1")
  .withRegion(Region.EUROPE_WEST)
  .withExistingResourceGroup(group)
  .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2)
  .withExistingPrimaryNetworkSubnet(network, subnet)
  .withoutPrimaryInternetFacingLoadBalancer()
  .withoutPrimaryInternalLoadBalancer()
  .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
  .withRootUsername("secret")
  .withRootPassword("itsNOTsecret12345")
  .withoutOverProvisioning()
  .withCapacity(1)
  .createAsync();

final SettableFuture<VirtualMachineScaleSet> future = SettableFuture.create();
futures.add(future);
Utils.<VirtualMachineScaleSet>rootResource(vms).subscribe(new Action1<VirtualMachineScaleSet>() {
  public void call(VirtualMachineScaleSet arg0) {
    future.set(arg0);
  }
}, new Action1<Throwable>() {
  public void call(Throwable arg0) {
    System.err.println(arg0);
    future.set(null);        
  }      
});
    
Observable<Indexable> vms2 = azure.virtualMachineScaleSets()
  .define("test2")
  .withRegion(Region.EUROPE_WEST)
  .withExistingResourceGroup(group)
  .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2)
  .withExistingPrimaryNetworkSubnet(network, subnet)
  .withoutPrimaryInternetFacingLoadBalancer()
  .withoutPrimaryInternalLoadBalancer()
  .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
  .withRootUsername("secret")
  .withRootPassword("itsNOTsecret12345")
  .withoutOverProvisioning()
  .withCapacity(2)
  .createAsync();

final SettableFuture<VirtualMachineScaleSet> future2 = SettableFuture.create();
futures.add(future2);
Utils.<VirtualMachineScaleSet>rootResource(vms2).subscribe(new Action1<VirtualMachineScaleSet>() {
  public void call(VirtualMachineScaleSet arg0) {
    future2.set(arg0);
  }
}, new Action1<Throwable>() {
  public void call(Throwable arg0) {
    System.err.println(arg0);
    future2.set(null);        
  }      
});

for (Future<VirtualMachineScaleSet> f : futures) {
  VirtualMachineScaleSet vmss = f.get();
  if (vmss != null) {
    System.out.println(vmss.name() + ": " + vmss.capacity());
  }
}
```

**Comment by anuchandy** [comment]
The code looks correct to me, one question - have you tried waiting for some time after the creation of the scale sets? I have seen that Azure starts with more instances than requested then it scale down to the exact number of instances.

**Comment by tgc123** [comment]
The case about getting "too many" instances in the beginning and scaling down later is related to [OverProvisioning](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-design-overview#overprovisioning). In this case the overprovisioning is switched off by using `.withoutOverProvisioning()`.

In this case waiting does not help, the instance count stays above the requested.

**Comment by anuchandy** [comment]
Thanks, @tgc123 didn't notice that overProvisioning is disabled explicitly. 

I tried to repo this but it seems working as expected i.e. it creates requested instance count. For your reference below is the response I received from the service for both VMSS PUT.

Initial response from compute service for first VMSS request

```json
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 1
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test1-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/{sub-id}/resourceGroups/javacsmrg37686/providers/Microsoft.Network/virtualNetworks/vmssvnet/subnets/subnet1"}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "f82b0351-6888-40af-80fb-9b51b220ae01"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "eastus",
  "tags": {},
  "id": "/subscriptions/{sub-id}/resourceGroups/javacsmrg37686/providers/Microsoft.Compute/virtualMachineScaleSets/test1",
  "name": "test1"
}
```

Initial response from compute service for second VMSS request

```json
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 2
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test2-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/{sub-id}/resourceGroups/javacsmrg37686/providers/Microsoft.Network/virtualNetworks/vmssvnet/subnets/subnet1"}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "67b5089d-0df4-4562-a513-27d642b4a952"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "eastus",
  "tags": {},
  "id": "/subscriptions/{sub-id}/resourceGroups/javacsmrg37686/providers/Microsoft.Compute/virtualMachineScaleSets/test2",
  "name": "test2"
}
```

Not sure why it's different in your case. Could you try enable logging and take a look at request response payload?

```java
            Azure azure = Azure.configure()
                    .withLogLevel(LogLevel.BODY_AND_HEADERS)
                    .authenticate(credFile)
                    .withDefaultSubscription();
```

**Comment by tgc123** [comment]
Thanks! In my log entries, both of the requests have capacity=2:

First request:
```
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 2
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test1-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"<removed>"}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "278a55b3-4498-47e3-b227-29d28ccbd3d7"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "westeurope",
  "tags": {},
  "id": "<removed>/providers/Microsoft.Compute/virtualMachineScaleSets/test1",
  "name": "test1"
}
```

Second request:
```
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 2
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test2-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"<removed>"}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "25909a21-91ff-4df2-afeb-6f1f46f3fff4"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "westeurope",
  "tags": {},
  "id": "<removed>/providers/Microsoft.Compute/virtualMachineScaleSets/test2",
  "name": "test2"
}
```

I'm running this in Windows Server 2012R2 using Eclipse with JDK1.8.0_102. Azure SDK is 1.0.0-beta5.

**Comment by anuchandy** [comment]
Thanks, ok so this is the response you are getting for the concurrent request, could you share the request log as well (request with HTTP method as PUT)?

**Comment by tgc123** [comment]
Sure, here's the request:
```
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> PUT https://management.azure.com/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachineScaleSets/test2?api-version=2016-04-30-preview
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-language: en-US
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - authorization: Bearer <removed>
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - user-agent: Azure-SDK-For-Java/1.0.0-beta5 (ComputeManagementClient, )
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - x-ms-client-request-id: 72ab4296-6b61-4bbb-9b68-56b5e71cc717
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-type: application/json; charset=utf-8
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-length: 1021
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - host: management.azure.com
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - connection: Keep-Alive
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-encoding: gzip
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - 1021-byte body:
{"location":"westeurope","tags":{},"sku":{"name":"Standard_D1_v2","tier":"Standard","capacity":2},"properties":{"upgradePolicy":{"mode":"Automatic"},"virtualMachineProfile":{"osProfile":{"computerNamePrefix":"test2-vm","adminUsername":"secret","adminPassword":"itsNOTsecret12345","linuxConfiguration":{"disablePasswordAuthentication":false}},"storageProfile":{"imageReference":{"publisher":"Canonical","offer":"UbuntuServer","sku":"16.04.0-LTS","version":"latest"},"osDisk":{"caching":"ReadWrite","createOption":"fromImage","managedDisk":{"storageAccountType":"Standard_LRS"}}},"networkProfile":{"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Network/virtualNetworks/<network>/subnets/<subnet>"},"loadBalancerBackendAddressPools":[],"loadBalancerInboundNatPools":[]}}]}}]}},"overprovision":false}}
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> END PUT
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> PUT https://management.azure.com/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachineScaleSets/test1?api-version=2016-04-30-preview
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-language: en-US
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - authorization: Bearer <removed>
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - user-agent: Azure-SDK-For-Java/1.0.0-beta5 (ComputeManagementClient, )
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - x-ms-client-request-id: 24d0ca8d-e9e9-45b4-9467-0acbe85a3b80
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-type: application/json; charset=utf-8
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-length: 1021
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - host: management.azure.com
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - connection: Keep-Alive
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-encoding: gzip
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - 1021-byte body:
{"location":"westeurope","tags":{},"sku":{"name":"Standard_D1_v2","tier":"Standard","capacity":2},"properties":{"upgradePolicy":{"mode":"Automatic"},"virtualMachineProfile":{"osProfile":{"computerNamePrefix":"test1-vm","adminUsername":"secret","adminPassword":"itsNOTsecret12345","linuxConfiguration":{"disablePasswordAuthentication":false}},"storageProfile":{"imageReference":{"publisher":"Canonical","offer":"UbuntuServer","sku":"16.04.0-LTS","version":"latest"},"osDisk":{"caching":"ReadWrite","createOption":"fromImage","managedDisk":{"storageAccountType":"Standard_LRS"}}},"networkProfile":{"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Network/virtualNetworks/<network>/subnets/<subnet>"},"loadBalancerBackendAddressPools":[],"loadBalancerInboundNatPools":[]}}]}}]}},"overprovision":false}}
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> END PUT
```

If I add a short sleep before creating the second ScaleSet, like:
```
...
Thread.sleep(500); // 500ms seems to be safe, 100ms is not enough
Observable<Indexable> vms2 = a.virtualMachineScaleSets()
...
```
Then the requests and responses looks correct:
```
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> PUT https://management.azure.com/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachineScaleSets/test1?api-version=2016-04-30-preview
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-language: en-US
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - authorization: Bearer <removed>
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - user-agent: Azure-SDK-For-Java/1.0.0-beta5 (ComputeManagementClient, )
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - x-ms-client-request-id: 830cd5a5-5dbe-4efa-bd77-4dfd89b51eae
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-type: application/json; charset=utf-8
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-length: 1022
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - host: management.azure.com
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - connection: Keep-Alive
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-encoding: gzip
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - 1022-byte body:
{"location":"westeurope","tags":{},"sku":{"name":"Standard_D1_v2","tier":"Standard","capacity":1},"properties":{"upgradePolicy":{"mode":"Automatic"},"virtualMachineProfile":{"osProfile":{"computerNamePrefix":"test1-vm","adminUsername":"secret","adminPassword":"itsNOTsecret12345","linuxConfiguration":{"disablePasswordAuthentication":false}},"storageProfile":{"imageReference":{"publisher":"Canonical","offer":"UbuntuServer","sku":"16.04.0-LTS","version":"latest"},"osDisk":{"caching":"ReadWrite","createOption":"fromImage","managedDisk":{"storageAccountType":"Standard_LRS"}}},"networkProfile":{"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Network/virtualNetworks/<network>/subnets/<subnet>"},"loadBalancerBackendAddressPools":[],"loadBalancerInboundNatPools":[]}}]}}]}},"overprovision":false}}
[RxIoScheduler-2] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> END PUT
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> PUT https://management.azure.com/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachineScaleSets/test2?api-version=2016-04-30-preview
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-language: en-US
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - authorization: Bearer <removed>
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - user-agent: Azure-SDK-For-Java/1.0.0-beta5 (ComputeManagementClient, )
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - x-ms-client-request-id: bb91792b-9dd9-4cc0-8a8f-f9c6e477289e
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-type: application/json; charset=utf-8
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - content-length: 1022
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - host: management.azure.com
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - connection: Keep-Alive
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - accept-encoding: gzip
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - 1022-byte body:
{"location":"westeurope","tags":{},"sku":{"name":"Standard_D1_v2","tier":"Standard","capacity":2},"properties":{"upgradePolicy":{"mode":"Automatic"},"virtualMachineProfile":{"osProfile":{"computerNamePrefix":"test2-vm","adminUsername":"secret","adminPassword":"itsNOTsecret12345","linuxConfiguration":{"disablePasswordAuthentication":false}},"storageProfile":{"imageReference":{"publisher":"Canonical","offer":"UbuntuServer","sku":"16.04.0-LTS","version":"latest"},"osDisk":{"caching":"ReadWrite","createOption":"fromImage","managedDisk":{"storageAccountType":"Standard_LRS"}}},"networkProfile":{"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":"/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Network/virtualNetworks/<network>/subnets/<subnet>"},"loadBalancerBackendAddressPools":[],"loadBalancerInboundNatPools":[]}}]}}]}},"overprovision":false}}
[RxIoScheduler-3] INFO com.microsoft.azure.management.compute.VirtualMachineScaleSets createOrUpdate - --> END PUT
```
Response 1:
```
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 1
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test1-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":<removed>}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "d9e9132b-4fab-430b-afb6-16ac9af08146"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "westeurope",
  "tags": {},
  "id": "<removed>/providers/Microsoft.Compute/virtualMachineScaleSets/test1",
  "name": "test1"
}
```
Response 2:
```
{
  "sku": {
    "name": "Standard_D1_v2",
    "tier": "Standard",
    "capacity": 2
  },
  "properties": {
    "singlePlacementGroup": true,
    "upgradePolicy": {
      "mode": "Automatic"
    },
    "virtualMachineProfile": {
      "osProfile": {
        "computerNamePrefix": "test2-vm",
        "adminUsername": "secret",
        "linuxConfiguration": {
          "disablePasswordAuthentication": false
        },
        "secrets": []
      },
      "storageProfile": {
        "osDisk": {
          "createOption": "FromImage",
          "caching": "ReadWrite",
          "managedDisk": {
            "storageAccountType": "Standard_LRS"
          }
        },
        "imageReference": {
          "publisher": "Canonical",
          "offer": "UbuntuServer",
          "sku": "16.04.0-LTS",
          "version": "latest"
        }
      },
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"primary-nic-cfg","properties":{"primary":true,"ipConfigurations":[{"name":"primary-nic-ip-cfg","properties":{"subnet":{"id":<removed>}}}]}}]}
    },
    "provisioningState": "Creating",
    "overprovision": false,
    "uniqueId": "51ef31e9-cd1c-4579-952c-a98a868076fa"
  },
  "type": "Microsoft.Compute/virtualMachineScaleSets",
  "location": "westeurope",
  "tags": {},
  "id": "<removed>/providers/Microsoft.Compute/virtualMachineScaleSets/test2",
  "name": "test2"
}
```

**Comment by anuchandy** [comment]
@tgc123 thank you!. 

Found the issue, it is due to sharing the sku object internally between two VMSS creatables. I will push fix to the master this week and update this issue thread. Assigning this to me.

**Comment by tgc123** [comment]
Thanks, great to hear that!

**Comment by anuchandy** [comment]
Fix is merged and available in v > 1.0.0
